rook-source-mysql/src/main/java/com/github/shyiko/rook/source/mysql/MySQLReplicationStream.java

```diff
@@ -80,16 +84,18 @@ public void setGroupEventsByTX(boolean groupEventsByTX) {
         this.groupEventsByTX = groupEventsByTX;
     }
 
-    public void setSuppressHeartbeatEvents(boolean suppressHeartbeatEvents) {
-        this.suppressHeartbeatEvents = suppressHeartbeatEvents;
+    public void setExceptionHandler(ReplicationEventExceptionHandler exceptionHandler) {
+        this.exceptionHandler = exceptionHandler;
     }
 
-    public void setHeartbeatTable(String heartbeatTable) {
-        this.heartbeatTable = heartbeatTable;
+    public void setIgnoredHostsIds(Set<Long> ignoredHostIds) {
+        this.ignoredHostIds = ignoredHostIds;
+        filteringEnabled = this.ignoredHostIds != null && !this.ignoredHostIds.isEmpty();
```

Comment:
@igrunskiy setIgnoredHostsIds([1L, ...]) && setIgnoredTables(null) results in former being ignored which is counterintuitive (vice versa may result in NPE down the pipe). I would recommend to remove flag & not check for null.